lineman/app/models/discussion_model.coffee

```diff
@@ -9,6 +9,7 @@ angular.module('loomioApp').factory 'DiscussionModel', (BaseModel, AppConfig) ->
     afterConstruction: ->
       if @isNew()
         @private = @privateDefaultValue()
+      @volume = @discussionReaderVolume or @recordStore.memberships.find({userId: AppConfig.currentUserId, groupId: @groupId })[0].volume
```

Comment:
I looked at making it a function, but found it to be a little hairy because
in places we're doing things like `thread.volume = 'quiet'`, and I'd rather
deal with this problem than that problem.

That being said, not sure how to handle an updated membership just yet.

On Tue, Oct 27, 2015 at 10:56 PM, Robert Guthrie <notifications@github.com>
wrote:

> In lineman/app/models/discussion_model.coffee
> <https://github.com/loomio/loomio/pull/2649#discussion_r43185699>:
>
> > @@ -9,6 +9,7 @@ angular.module('loomioApp').factory 'DiscussionModel', (BaseModel, AppConfig) ->
> >      afterConstruction: ->
> >        if @isNew()
> >          @private = @privateDefaultValue()
> > +      @volume = @discussionReaderVolume or @recordStore.memberships.find({userId: AppConfig.currentUserId, groupId: @groupId })[0].volume
>
> At first look I think this should be a function.. but now I understand
> you're doing it so we can query on this. Seems like the right thing to do,
> but if membership volume changes, what do we do?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/loomio/loomio/pull/2649/files#r43185699>.
>